src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -244,6 +249,31 @@ internal sealed class CompilerAnalyzerAssemblyResolver(AssemblyLoadContext compi
                 }
             }
         }
+
+        private sealed class DiskResolver : IAnalyzerAssemblyResolver
+        {
+            public static readonly DiskResolver Instance = new DiskResolver();
```

Comment:
nit:

```suggestion
            public static readonly DiskResolver Instance = new DiskResolver();
            
```

Nit: is it important to expose this as an actual DiskResolver?  Would is suffice to expose as an IAnalyzerAssemblyResolver?  I personally prefer the latter as then someone can't start calling new APIs directly on this.

Reply:
It's a `private` nested type so didn't see much risk in the new APIs being called.